### PR TITLE
✨ [API-2179] Add Lido Satellite Swap Adapter

### PR DESCRIPTION
**Pull request by foxpy**

### CODE REVIEW

**Review by ratik — APPROVED** [review]

File: contracts/adapters/swap/lido-satellite/src/contract.rs
Comment (NotJeremyLiu) [inline contracts/adapters/swap/lido-satellite/src/contract.rs:92]: If this is immutable on the lido side, do you think it makes sense to add the bridge_denom and canonical_denom in the instantiation method instead, saving us an external contract query per call? (as well as making testing a tad easier)

If there is any expectation this is not immutable, I'd err on the side of keeping this as is to avoid any future issues

**Review by NotJeremyLiu — COMMENTED** [review]
If you could add unit tests attempting to follow the same structure as the existing swap adapter ones that would be great, (just checks the msg created is as expected, doesn't need to use cw-multi-test).

And although may never be used, adding the query methods for completeness would be nice

File: contracts/adapters/swap/lido-satellite/src/contract.rs
Comment (foxpy) [reply contracts/adapters/swap/lido-satellite/src/contract.rs:92]: I've actually been thinking about this one! Yes, I like the idea of caching these values during the instantiation.

While canonical denom will never change, we plan to abolish bridged denom eventually, so this adapter will stop working anyway no matter if we cache denoms or not. So, yeah, let's just cache them. I will implement it shortly.

**Comment by foxpy** [comment]
Alright, I have implemented denom caching, unit tests and a very basic query stub. The latter should probably be improved, I am not really sure if we have to be bothered about it.

**Review by NotJeremyLiu — APPROVED** [review]

**Comment by NotJeremyLiu** [comment]
Updated for cw20 support refactor, tested on neutron mainnet for both mint and burn, approving. Will merge once cw20 support is merged into main.

@foxpy thank you for your awesome work here!

**Review by zrbecker — APPROVED** [review]

**Pull request merged**
